strawberry/type.py

```diff
@@ -92,7 +94,10 @@ def _get_fields(wrapped):
             if hasattr(klass, IS_STRAWBERRY_INTERFACE)
         ]
 
-    wrapped = dataclasses.dataclass(cls)
+        # the user might want to register other classes,
+        # but for now let's just add our dataclass
+        extra_kwargs["is_type_of"] = lambda obj, _: isinstance(obj, wrapped)
```

Comment:
can wee expand a bit on this? not sure what you mean with that comment :)

Reply:
Basically the user could return other datatypes (like an instance of an ORM class, a named tuple) instead of explicitly wrapping the data in an instance of the dataclass, right?
But this would not work for interfaces unless `is_type_of` called on GraphQLObjectType returns `True` for the used class. This could be done in various ways, from additional "meta" attributes on the dataclass to arguments to the decorator and there either as another lambda function or as a list of types. Since I didn't want to make this decision yet I only wanted to add a comment...

Would the following explain it better?
```
-        # the user might want to register other classes,
-        # but for now let's just add our dataclass
+        # make sure instances of our wrapped dataclass
+        # are recognized by the GraphQLObjectType as types with the correct interface 
+        # TODO: user might want to add custom types (like ORM classes) at some point
```

Reply:
ah I see! I think we can keep the comment like that, the context on this PR should be enough for now :)